doc/release-notes.md

```diff
@@ -19,44 +20,239 @@ dashd/dash-qt (on Linux).
 
-- Fixed crash when processing invalid masternode payment destinations, replacing unsafe assertion with proper error handling (dash#6740).
+Breaking Change: Block Filter Index Format Update
+-------------------------------------------------
 
-RPC and Logging Improvements
-----------------------------
+The compact block filter index format has been updated to include Dash special transaction data, providing feature parity with bloom filters for SPV client support. This change is incompatible with existing blockfilter indexes. Existing blockfilter indexes will automatically be re-created with the new version.
 
-- Fixed misleading error logs that were triggered by legitimate RPC queries for non-existent transaction data, reducing log noise and preventing false alarms (dash#6744).
+- The blockfilter index now includes fields from Dash special transactions:
+  - ProRegTx (masternode registration)
+  - ProUpServTx (masternode service updates)
+  - ProUpRegTx (masternode operator updates)
+  - ProUpRevTx (masternode revocation)
+  - AssetLockTx (platform credit outputs)
+- A versioning system has been added to detect incompatible indexes on startup
+- The index version is now 2 (previously unversioned)
 
-Performance Improvements
-------------------------
+### Benefits
 
-- Optimized versionbits calculation to avoid unnecessary computations during block operations, significantly improving performance during blockchain reorganizations (dash#6632).
+- SPV clients can now detect and track Dash-specific transactions
+- Feature parity between bloom filters and compact block filters
+- Protection against serving incorrect filter data to light clients
 
-Documentation Updates
+Other notable changes
 ---------------------
 
-- Updated translation documentation with current Transifex links and fixed typos to help contributors properly access translation resources (dash#6739).
+* Dash Core will no longer permit the registration of new legacy scheme masternodes after the deployment of the v24
+  fork. Existing basic scheme masternodes will also be prohibited from downgrading to the legacy scheme after the
+  deployment is active. (#6729)
+* To help prevent fingerprinting transactions created by the Dash Core wallet, change output
+  amounts are now randomized. (#6685)
+* Dash Core will no longer migrate EvoDb databases generated in v19 and v20, users upgrading
+  from these versions are recommended to run `-reindex` to rebuild all databases and indexes. (#6579)
+
+Updated REST APIs
+-----------------
+
+- The `/headers/` and `/blockfilterheaders/` endpoints have been updated to use
+  a query parameter instead of path parameter to specify the result count. The
+  count parameter is now optional, and defaults to 5 for both endpoints. The old
+  endpoints are still functional, and have no documented behaviour change. (#24098)
```

Comment:
```suggestion
- The `/headers/` and `/blockfilterheaders/` endpoints have been updated to use
  a query parameter instead of a path parameter to specify the result count. The
  count parameter is now optional and defaults to 5 for both endpoints. The old
  endpoints are still functional, and have no documented behaviour change. (#24098)
```